LobbyManager.h

```diff
@@ -48,11 +49,12 @@ class LobbyManager {
 		if (active_lobbies.find(id) == active_lobbies.end()) {
 			throw std::runtime_error("No active lobby with this ID");
 		} 
-		return active_lobbies.at(id);
+		auto requested_lobby = active_lobbies.at(id);
+		return {requested_lobby.player1, requested_lobby.player2, requested_lobby.search_id};
```

Comment:
Again, this isn't what I meant with my previous comment. The fields specifying whether the players are active are still publicly accessible outside this class.

Reply:
Again, you haven't changed the root of the issue. In fact, you've made it worse. You have moved information designated to the lobby to the `PlayerInfo` struct which doesn't need it.